vividus-plugin-web-app/src/main/java/org/vividus/ui/web/action/JavascriptActions.java

```diff
@@ -163,7 +164,21 @@ public String getUserAgent()
     @Override
     public double getDevicePixelRatio()
     {
-        return devicePixelRatio.get();
+        double dpr;
+        if ((boolean) executeScript("return window != window.top"))
+        {
+            String href = getFrameHref();
+            webDriverProvider.get().switchTo().defaultContent();
+            dpr = devicePixelRatio.get();
+            WebElement frame = webDriverProvider.get()
+                    .findElement(By.xpath(String.format("//iframe[@src=\"%s\"]", href)));
+            webDriverProvider.get().switchTo().frame(frame);
```

Comment:
can't we use a plain JS to extract DPR?

Reply:
I think no, because of switching between iframes documents with a different script origin (https://docs.w3cub.com/dom/window/frameelement/)